### ISSUE TITLE
Incorrect/useless UsernameAllowedChars check

### ISSUE THREAD

**Issue by MishimaHaruna** [HerculesWS/FluxCP#34]
The check for `UsernameAllowedChars` is incorrect, making it completely useless.

See `lib/Flux/LoginServer.php`:
```
if (preg_match('/^[^' . Flux::config('UsernameAllowedChars') . ']$/', $username)) {
	throw new Flux_RegisterError('Invalid character(s) used in username', Flux_RegisterError::INVALID_USERNAME);
}
```
The regex will never match, unless the username is exactly one character long.

This is likely to be present in all FluxCP forks, including the original project (Google Code).

Our repository has lost track of the originating commit due to missxantara's mis-handling of upstream merges (i.e. merge of over 200 commits at once at 3427dbb49f65012b90a232962533e9f664a944fa), but it was in Paradox924X's SVN revision 1106 (diff at https://code.google.com/p/fluxcp/source/diff?spec=svn1133&r=1106&format=side&path=/branches/fluxcp-1.0/lib/Flux/LoginServer.php while Google Code is still alive).
It's clear from that commit that `Modified patterns for username and email address validation during registration to employ start-of-string and end-of-string anchors.` is probably a bad idea, unless one understand regex properly.

**Comment by Streusel** [comment]
Could this be a preventative measure just like how users are able to create 1 character long character?
e.g. A(<some microsoft key combination>x4)

**Comment by MishimaHaruna** [comment]
It isn't. If the string is one character long, it'd be stopped by `MinUsernameLength`. If the string is longer, well, it's longer and it won't be matched by this regex.

**Comment by MishimaHaruna** [comment]
This was fixed in a904a24a6f10e28b14f37f694cc09ed76da1e68d
